### PR TITLE
feat(datafusion): Add Timestamp scalar value conversion for predicate pushdown

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

- Closes #.

## What changes are included in this PR?

Add support for converting DataFusion Timestamp scalar values to Iceberg Datum for predicate pushdown. This enables timestamp literal comparisons to be pushed down to the storage layer when DataFusion provides timestamp literals in ScalarValue form, improving query performance.

Changes:
- Add conversion for 2 timestamp time units (Microsecond, Nanosecond) to Iceberg's native representations
- Add comprehensive unit tests covering all time units and edge cases
- Add sqllogictest (timestamp_predicate_pushdown.slt) to validate end-to-end timestamp predicate filtering and pushdown behavior

Implementation details:
- Iceberg uses microseconds and nanoseconds as the native timestamp representations
- Timezone information in ScalarValue is preserved but not used in conversion (Iceberg timestamp type is timezone-agnostic)

### CODE REVIEW

File: crates/sqllogictest/src/engine/datafusion.rs [hunk 1]
```diff
@@ -96,6 +96,7 @@ impl DataFusionEngine {
         // Create partitioned test table (unpartitioned tables are now created via SQL)
         Self::create_partitioned_table(&catalog, &namespace).await?;
         Self::create_binary_table(&catalog, &namespace).await?;
+        Self::create_timestamp_table(&catalog, &namespace).await?;
```
Comment (liurenjie1024) [inline crates/sqllogictest/src/engine/datafusion.rs]: We already supported creating table, see https://github.com/apache/iceberg-rust/blob/045c0b2e298fee03e6d658bba78092a7fcf65747/crates/sqllogictest/testdata/slts/df_test/basic_queries.slt#L23, we don't need to create table using rust code.

File: crates/integrations/datafusion/src/physical_plan/expr_to_predicate.rs [hunk 2]
```diff
@@ -266,6 +270,17 @@ fn scalar_value_to_datum(value: &ScalarValue) -> Option<Datum> {
...
         ScalarValue::Date64(Some(v)) => Some(Datum::date((*v / MILLIS_PER_DAY) as i32)),
+        // Timestamp conversions - convert to microseconds (Iceberg's native timestamp unit)
+        ScalarValue::TimestampSecond(Some(v), _) => {
+            Some(Datum::timestamp_micros(v * MICROS_PER_SECOND))
+        }
+        ScalarValue::TimestampMillisecond(Some(v), _) => {
+            Some(Datum::timestamp_micros(v * MICROS_PER_MILLI))
+        }
+        ScalarValue::TimestampMicrosecond(Some(v), _) => Some(Datum::timestamp_micros(*v)),
+        ScalarValue::TimestampNanosecond(Some(v), _) => {
```
Comment (liurenjie1024) [inline crates/integrations/datafusion/src/physical_plan/expr_to_predicate.rs]: We should not convert nano seconds here since it will lose precision? For example, let's say the predict is like `c1 >= 100 nano seconds`, this will be converted to `c1 >= 0 micro seconds`, where `c1`'s data type is microseconds timestamp in iceberg. In this case if c1 = 0, it will not pass first condition, while it will pass second condition.

**Comment by liurenjie1024** [comment]
The ci failure should have been fixed.

**Review by blackmwk — APPROVED** [review]
Thanks @viirya for this pr!

**Comment by viirya** [comment]
Thanks @liurenjie1024 @blackmwk

**Comment by colinmarc** [comment]
I want to note that in my testing, this is broken on date-based pushdowns (extremely common in practice) without the extra conversions for `date` in https://github.com/apache/iceberg-rust/pull/1569.
